### PR TITLE
Add parsers to `WedLog`

### PR DESCRIPTION
**Pull request by wasjoe1**
Changes:
**app**
=> Edit the addressbook command parser
=> Add the guest and vendor command parsers
=> Add all the command parsers and respective command classes (only list has no parser)
=> command classes has empty constructors and return null upon execution (delete commands old code was left as comments)
=> Edit parserUtil to parse Dietary & RSVP requirements
=> No change to ModelManager (initial changes were removed)

**tests**
=> Edit the AddressBook parser test
=> Add the guest and vendor command parsers test
=> Edit parserUtilTest to parse Dietary & RSVP requirements
=> Add respective command parsers tests (VendorAddParsersTest & GuestAddParsersTest are empty due to the uncertainty of how we want to implement optional fields)

### CODE REVIEW

**Comment by wasjoe1** [comment]
Formatting errors noticed, will resolve asap

File: src/main/java/seedu/address/logic/commands/GuestAddCommand.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import seedu.address.model.person.Person;
+
+/**
+ * Adds a Guest to the address book.
+ */
+public class GuestAddCommand extends Command {
+    // below implementation more or less copied from AddCommand
+    public static final String COMMAND_WORD = "add";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds a person to the address book. ";
```
Comment (samuelim01) [inline src/main/java/seedu/address/logic/commands/GuestAddCommand.java:14]: Perhaps you could add the command parameters?

```suggestion
    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds a guest to the address book. "
            + "Parameters: "
            + PREFIX_NAME + "NAME "
            + "[" + PREFIX_PHONE + "PHONE] "
            + "[" + PREFIX_EMAIL + "EMAIL] "
            + "[" + PREFIX_ADDRESS + "ADDRESS] "
            + "[" + PREFIX_DIETARY + "DIETARY] "
            + "[" + PREFIX_RSVP + "RSVP] "
            + "[" + PREFIX_TAG + "TAG]...\n"
            + "Example: " + COMMAND_WORD + " "
            + PREFIX_NAME + "John Doe "
            + PREFIX_PHONE + "98765432 "
            + PREFIX_EMAIL + "johnd@example.com "
            + PREFIX_ADDRESS + "311, Clementi Ave 2, #02-25 "
            + PREFIX_DIETARY + "no beef"
            + PREFIX_RSVP + "no"
            + PREFIX_TAG + "friends "
            + PREFIX_TAG + "owesMoney";
            
```

File: src/main/java/seedu/address/logic/commands/GuestAddCommand.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    public static final String MESSAGE_SUCCESS = "New person added: %1$s";
+
+    public GuestAddCommand(Person person) {
+        // temporary empty constructor
```
Comment (samuelim01) [inline src/main/java/seedu/address/logic/commands/GuestAddCommand.java:18]:
```suggestion
        // TODO: initialize GuestAddCommand
```

**Review by samuelim01 — CHANGES_REQUESTED** [review]
LGTM, just a few nits!

File: src/main/java/seedu/address/logic/commands/GuestAddCommand.java (hunk 2 above)
Comment (wasjoe1) [reply src/main/java/seedu/address/logic/commands/GuestAddCommand.java:18]: will  be done in the next evolve!

File: src/main/java/seedu/address/logic/commands/GuestAddCommand.java (hunk 1 above)
Comment (wasjoe1) [reply src/main/java/seedu/address/logic/commands/GuestAddCommand.java:14]: will  be done in the next evolve!

File: src/test/java/seedu/address/logic/parser/GuestCommandParserTest.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+import seedu.address.logic.commands.GuestViewCommand;
+import seedu.address.logic.commands.HelpCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+public class GuestCommandParserTest {
+    private static final String GUEST_ADD_PARSE_EXCEPTION_MSG =
+            "Guest not created in GuestAddCommand due to un-evolved classes";
+    private GuestCommandParser parser = new GuestCommandParser();
+
+    // test guest add
```
[the remark this answers is not in the document]
Comment (wasjoe1) [reply src/test/java/seedu/address/logic/parser/GuestCommandParserTest.java]: resolved!

File: src/main/java/seedu/address/logic/parser/VendorCommandParser.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+     * @param userInput full user input string
+     * @return the command based on the user input
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public Command parseCommand(String userInput) throws ParseException {
+        final Matcher matcher = BASIC_COMMAND_FORMAT.matcher(userInput.trim()); // trims the back and end of the string
+        if (!matcher.matches()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, HelpCommand.MESSAGE_USAGE));
+        }
+        // any way the "vendor command is removed, should have the same format as previous"
```
[the remark this answers is not in the document]
Comment (wasjoe1) [reply src/main/java/seedu/address/logic/parser/VendorCommandParser.java:42]: resolved!

File: src/test/java/seedu/address/logic/parser/AddressBookParserTest.java [hunk 5]
```diff
@@ -1,99 +1,46 @@
...
-        List<String> keywords = Arrays.asList("foo", "bar", "baz");
-        FindCommand command = (FindCommand) parser.parseCommand(
-                FindCommand.COMMAND_WORD + " " + keywords.stream().collect(Collectors.joining(" ")));
-        assertEquals(new FindCommand(new NameContainsKeywordsPredicate(keywords)), command);
-    }
-
-    @Test
-    public void parseCommand_help() throws Exception {
-        assertTrue(parser.parseCommand(HelpCommand.COMMAND_WORD) instanceof HelpCommand);
-        assertTrue(parser.parseCommand(HelpCommand.COMMAND_WORD + " 3") instanceof HelpCommand);
```
[the remark this answers is not in the document]
Comment (wasjoe1) [reply src/test/java/seedu/address/logic/parser/AddressBookParserTest.java]: resolved!

File: src/main/java/seedu/address/logic/parser/GuestAddCommandParser.java [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+                : ParserUtil.parseAddress(argMultimap.getValue(PREFIX_ADDRESS).get());
+        RsvpStatus rsvpStatus = argMultimap.getValue(PREFIX_RSVP).isEmpty() // this is whether Optional is empty
+                ? null
+                : ParserUtil.parseRsvp(argMultimap.getValue(PREFIX_RSVP).get());
+        DietaryRequirements dietaryRequirements = argMultimap.getValue(PREFIX_DIETARY).isEmpty()
+                ? null
+                : ParserUtil.parseDietary(argMultimap.getValue(PREFIX_DIETARY).get());
+        Set<Tag> tagList = argMultimap.getAllValues(PREFIX_TAG).isEmpty() // checks if list of strings is empty
+                ? null
+                : ParserUtil.parseTags(argMultimap.getAllValues(PREFIX_TAG));
```
[the remark this answers is not in the document]
Comment (wasjoe1) [reply src/main/java/seedu/address/logic/parser/GuestAddCommandParser.java]: resolved!

**Review by samuelim01 — APPROVED** [review]
Nice work, LGTM!
